src/MagnumPlugins/UfbxImporter/UfbxImporter.h

```diff
@@ -0,0 +1,364 @@
+target:
+
+@code{.cmake}
+find_package(MagnumPlugins REQUIRED UfbxImporter)
+
+# ...
+target_link_libraries(your-app PRIVATE MagnumPlugins::UfbxImporter)
+@endcode
+
+See @ref building-plugins, @ref cmake-plugins, @ref plugins and
+@ref file-formats for more information.
+
+@section Trade-UfbxImporter-behavior Behavior and limitations
+
+The plugin supports @ref ImporterFeature::OpenData, @relativeref{ImporterFeature,OpenState} and @relativeref{ImporterFeature,FileCallback} features.
+Immediate dependencies are loaded during the initial import meaning the
+callback is called with @ref InputFileCallbackPolicy::LoadTemporary.
+In case of images, the files are loaded on-demand inside @ref image2D() calls
+with @ref InputFileCallbackPolicy::LoadTemporary and
+@ref InputFileCallbackPolicy::Close is emitted right after the file is fully
+read.
+
+The importer recognizes @ref ImporterFlag::Verbose if built in debug mode
+(@ref CORRADE_IS_DEBUG_BUILD or @cpp NDEBUG @ce if not defined). The verbose
+logging prints detailed ufbx-internal callstacks on load failure that can be
+used for debugging or reporting issues.
+
+@subsection Trade-UfbxImporter-behavior-scene Scene import
+
+-   ufbx supports only a single scene, though in practice it is extremely rare
+    to have FBX files containing more than a single scene.
+-   FBX files contain an implicit root node that is ignored unless the
+    @cb{.ini} preserveRootNode @ce @ref Trade-UfbxImporter-configuration "configuration option" is enabled.
+-   FBX files may contain nodes with "geometric transforms" that transform only
+    the mesh of the node without affecting children. These are converted to
+    unnamed helper nodes in the scene unless the @cb{.ini} geometricTransformNodes @ce
+    @ref Trade-UfbxImporter-configuration "configuration option" is explicitly disabled.
+-   Imported scenes always have @ref SceneMappingType::UnsignedInt, with
+    @ref SceneData::mappingBound() equal to @ref objectCount(). The scene is
+    always 3D.
+-   All reported objects have a @ref SceneField::Parent (of type
+    @ref SceneFieldType::Int), @ref SceneField::Translation (of type @ref SceneFieldType::Vector3d),
+    @ref SceneField::Rotation (of type @ref SceneFieldType::Quaterniond),
+    @ref SceneField::Scaling (of type @ref SceneFieldType::Vector3d) and
+    importer-specific flags @cpp "Visibility" @ce and @cpp "GeometricTransformHelper" @ce
+    (of type @ref SceneFieldType::UnsignedByte representing a boolean value).
+    These five fields share the same object mapping with
+    @ref SceneFieldFlag::ImplicitMapping set.
+-   Scene field @cpp "Visibility" @ce specifies whether objects should be visible
+    in some application-defined manner.
+-   Scene field @cpp "GeometricTransformHelper" @ce is set on synthetic nodes
+    that are not part of the original file, but represent FBX files' ability to
+    transform geometry without affecting children, see @ref Trade-UfbxImporter-scene-processing
```

Comment:
Marking this here so I don't forget to write this up as I go through the review comments later.